11ac kconfig update

abhinavnxp opened the pull request:
Following changes have been added - 

1. Added configuration to disable 11AC for 11n only SoC's similar to 11AX.
    - used this kconfig to disable 80211ac confs
    - disabled 11AX for IW416 since it is 11n soc.
 2. Added kconfig for regulatory domain in hostapd. Default is set to
    WW but changed it to default US for IW610 and RW610 socs.

krish2718 commented on modules/hostap/Kconfig:
```diff
@@ -596,4 +596,9 @@ config WIFI_NM_WPA_CTRL_RESP_TIMEOUT_S
 
+config WIFI_NM_HOSTAPD_REGULATORY_REGION
+	string "Select Wi-Fi Regulatory Domain"
+	default "WW "
```
Also the space at the end has a special meaning and may not be obvious to the user, I would suggest two separate configs for that similar to hostapd.conf https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf

```
config WIFI_NM_HOSTAPD_REGULATORY_REGION -> country[0], country[1], "country_code" in the config file
config WIFI_NM_HOSTAPD_REGULATORY_ENV -> country[2], "country3" in config file
```

abhinavnxp replied:
Addressed please check

nordicjm commented on boards/shields/nxp_m2_wifi_bt/Kconfig.defconfig:
```diff
@@ -55,7 +61,39 @@ endif # WIFI
+	default NXP_IW610_MURATA_2LL_M2
+endchoice
+
+if WIFI_NM_HOSTAPD_AP
+config WIFI_NM_HOSTAPD_REGULATORY_REGION
+	default "US "
+endif # WIFI_NM_HOSTAPD_AP
```
missing spacing, should be newline gaps, also default value has a space which looks like a mistake, also unsure if value will be accepted into e.g. menuconfig without it automatically removing space

abhinavnxp replied:
Addressed please check

nordicjm commented on soc/nxp/rw/Kconfig.defconfig.wifi:
```diff
@@ -15,3 +15,8 @@ config WIFI_MGMT_AP_MAX_NUM_STA
 endif # NET_L2_WIFI_MGMT
 endif # NETWORKING
+
+if WIFI_NM_HOSTAPD_AP
+config WIFI_NM_HOSTAPD_REGULATORY_REGION
+	default "US "
+endif # WIFI_NM_HOSTAPD_AP
```
as above

abhinavnxp replied:
Addressed please check

MaochenWang1 commented:
@abhinavnxp  please rebase your PR

nordicjm commented on boards/shields/nxp_m2_wifi_bt/Kconfig.defconfig:
```diff
@@ -24,6 +24,11 @@ choice NXP_IW416_MODULE
 	default NXP_IW416_MURATA_1XK_M2
 endchoice
 
+if WIFI_NM_HOSTAPD_AP
+config WIFI_NM_HOSTAPD_REGULATORY_REGION
+	default "WW"
+endif # WIFI_NM_HOSTAPD_AP
```
```suggestion
if WIFI_NM_HOSTAPD_AP

config WIFI_NM_HOSTAPD_REGULATORY_REGION
	default "WW"

endif # WIFI_NM_HOSTAPD_AP
```
fix in whole PR

abhinavnxp replied:
Addressed

nordicjm commented on soc/nxp/rw/Kconfig.defconfig.wifi:
```diff
@@ -15,3 +15,8 @@ config WIFI_MGMT_AP_MAX_NUM_STA
 endif # NET_L2_WIFI_MGMT
 endif # NETWORKING
+
+if WIFI_NM_HOSTAPD_AP
+config WIFI_NM_HOSTAPD_REGULATORY_REGION
+	default "US"
+endif # WIFI_NM_HOSTAPD_AP
```
is not fixed

krish2718 commented on modules/hostap/CMakeLists.txt:
```diff
@@ -213,6 +212,10 @@ zephyr_library_sources(
   ${HOSTAP_SRC_BASE}/utils/ip_addr.c
 )
 
+zephyr_library_sources_ifdef(CONFIG_WIFI_NM_WPA_SUPPLICANT_11AC
```
`CONFIG_WIFI_NM_WPA_HOSTAPD_11AC`?

abhinavnxp replied:
I have followed method similar to that of 11AX, is there any reason why do you expect it to be different?

krish2718 replied:
Well, looking at the code: `depends on WIFI_NM_WPA_SUPPLICANT_AP || WIFI_NM_HOSTAPD_AP`  so it's applicable for wpa_supplicant AP mode too. Let's leave it as is then.

abhinavnxp commented on boards/shields/nxp_m2_wifi_bt/Kconfig.defconfig:
```diff
@@ -16,6 +16,12 @@ if WIFI
 config WIFI_NXP
 	default y
 
+config WIFI_NM_WPA_SUPPLICANT_11AX
```
Already it is a separate commit am I missing anything here? or can you elaborate which changes you want separate commit and reason behind it?

krish2718 replied:
Each commit should have one logical change and preferably single module, so, in this commit it has boards + hostap changes, so, if you want revert in the future it becomes tough.

abhinavnxp replied:
Addressed

abhinavnxp commented on boards/shields/nxp_m2_wifi_bt/Kconfig.defconfig:
```diff
@@ -24,6 +24,13 @@ choice NXP_IW416_MODULE
 	default NXP_IW416_MURATA_1XK_M2
 endchoice
 
+if WIFI_NM_HOSTAPD_AP
```
Already it is a separate commit am I missing anything here? or can you elaborate which changes you want separate commit and reason behind it?

abhinavnxp committed:
modules: hostap: Updated reg domain details

Added kconfig for regulatory domain in hostapd. Default is set to
00 as World Wide mode.
Also added 3rd octet kconfig to set regulatory environment and set
it to 32 for all supported environments.

Signed-off-by: Abhinav Kulkarni <abhinav.kulkarni@nxp.com>

abhinavnxp committed:
boards: shield: nxp_m2_wifi_bt: Updated reg domain

Set regulatory domain to default US for IW610 and RW610 socs
and WW for IW416 and IW612 socs.

Signed-off-by: Abhinav Kulkarni <abhinav.kulkarni@nxp.com>

abhinavnxp committed:
modules: hostap: Added 11AC kconfig

Added 11AC kconfig to disable 11AC mode.
Used this kconfig to disable 80211ac confs.

Signed-off-by: Abhinav Kulkarni <abhinav.kulkarni@nxp.com>

abhinavnxp committed:
boards: shields: nxp_mw_wifi_bt: Disable modes for IW416

Disabled 11AX and 11AC for IW416 since it is 11n soc.

Signed-off-by: Abhinav Kulkarni <abhinav.kulkarni@nxp.com>
